gem/backends.py

```diff
@@ -36,15 +37,29 @@ def _update_user_from_claims(user, claims):
     user.email = claims.get("email", "")
     user.save()
 
+    username = claims.get("username", "")
+
     # If the user doesn't have a profile for some reason make one
     if not hasattr(user, 'profile'):
         user.profile = UserProfile(user=user)
+
+        # TODO: we should be using a more specific site here?
         user.profile.site = Site.objects.get(is_default_site=True)
 
     # Ensure the profile is linked to their auth service account using the uuid
     if user.profile.auth_service_uuid is None:
         user.profile.auth_service_uuid = claims.get("sub")
 
+        # If a user already exists with this username
+        # change that user's username
+        if username:
+            for u in User.objects.filter(
+                    username=username).exclude(pk=user.pk):
+                u.username = str(u.profile.site.pk) + '_' + username
```

Comment:
We should probably also check that this user doesn't already have an `auth_service_uuid` and throw the `FieldError` here if they do